src/sentry/sentry_apps/tasks/service_hooks.py

```diff
@@ -40,16 +41,24 @@ def get_payload_v0(event):
     taskworker_config=TaskworkerConfig(namespace=sentryapp_tasks, retry=Retry(times=3)),
 )
 @retry
-def process_service_hook(servicehook_id, event, **kwargs):
+def process_service_hook(
+    servicehook_id: int,
+    event: Any,
+    payload: str | None = None,
+    project_id: int | None = None,
+    **kwargs,
+):
     try:
         servicehook = ServiceHook.objects.get(id=servicehook_id)
     except ServiceHook.DoesNotExist:
         return
 
-    if servicehook.version == 0:
-        payload = get_payload_v0(event)
-    else:
-        raise NotImplementedError
+    project_id = project_id or event.project_id
+    if not payload:
+        if servicehook.version == 0:
+            payload = json.dumps(get_payload_v0(event))
+        else:
+            raise NotImplementedError
```

Comment:
Passing in the serialized payload as a string saves us pickle + two json.dumps calls here 🚀